docs/tutorials/first_steps/07_building_scripts.rst

```diff
@@ -0,0 +1,135 @@
+    
+        #
+        # Usage: python filter_flights.py <input-path> <output-path>
+        #
+    
+        import pandas as pd
+        import sys
+
+        # It would be more robust to use argparse or click, but we want this to be simple
+        if len(sys.argv) < 3:
+	    sys.exit("Please invoke with two arguments: input and output paths")
+
+        input_path = sys.argv[1]
+        output_path = sys.argv[2]
+
+
+        # Read in the data
+        df = pd.read_csv(input_path)
+
+        # Select only flights to Austin (AUS)
+        df = df[df['DEST'] == 'DFW']
+
+        # Save the result
+        df.to_csv(output_path, index=False)
+
+
+.. tabbed:: Julia
+    
+    `Download Julia script <https://renkulab.io/projects/renku-tutorial/flights-tutorial-julia/files/blob/.tutorial/meta/templates/FilterFlights.jl>`_
+    
+    .. code-block:: julia
+        
+        #
+        # Usage: julia FilterFlights.jl <input-path> <output-path>
+        #
+
+        using CSV, ZipFile, DataFrames
+
+        input_path = "../../data/flight-data/2019-01-flights.csv.zip"
+        output_path = "../../data/output/2019-01-flights-filtered.csv"
+        
+        if length(ARGS) > 0
+            input_path = ARGS[1]
+        end
+        if length(ARGS) > 1
+            output_path = ARGS[2]
+        end
+                
+        output_folder = dirname(output_path);
+        
+        # Read the flights database
+        println("Reading $input_path ...")
+        rows = CSV.File(ZipFile.Reader(input_path).files[1]);
+        full_df = rows |> DataFrame;
+        
+        # Filter to flights to Austin, TX
+        df = full_df[full_df.DEST .== "DFW", :]
+        
+        # Write the filtered list out
+        println("Writing $output_path ...")
+        run(`mkdir -p $output_folder`)
+        CSV.write(output_path, df)
+            
+.. tabbed:: R
+
+    `Download R script <https://renkulab.io/projects/renku-tutorial/flights-tutorial-r/files/blob/.tutorial/meta/templates/RunFilterFlights.R>`_
+
+    .. code-block:: r
+        
+        #
+        # Usage: Rscript RunFilterFlights.R <input-path> <output-path>
+        #
+        
+        args <- commandArgs(trailingOnly = TRUE)
+
+        if (length(args) != 2) {
+        stop("At least two arguments must be supplied (input and output files).",
+            call. = FALSE)
+        }
+        
+        inputPath <- args[1]
+        outputPath <- args[2]
+        
+        library(tidyverse)
+        
+        data <- unzip(inputPath) %>%
+        read.csv()
+        
+        filteredData <- data %>% filter(DEST == "DFW")
+        
+        outputFolder <- dirname(outputPath)
+        
+        if (!dir.exists(outputFolder)) {
+        dir.create(outputFolder)
+        }
+        
+        write.csv(filteredData, outputPath, row.names = FALSE)
+        
+    
+
+
+As before, you can download it and drag & drop into the JupyterLab session or
+upload the file using RStudio into your ``src`` directory. 
```

Comment:
I think we should recommend the structure `src/{python, julia, r}`.

Reply:
I'm not sure many projects would have mixtures of those types of files though? I feel users might be reluctant to put another layer of folder(s) in their repos if there'd only be one such folder in the src folder.

Reply:
I think that is your call, but then we should probably be consistent and apply the same structure to the python project as well.

FWIW, my projects usually start with just python, but I often end up having `bash` and `python` subfolders in my `src` directory and am glad I have them separate, but it is entirely possible I work differently from most users.

Reply:
I agree with being consistent throughout. I know that many of our more advanced collaborators have bash+_language_ folders to organise their src folder so it's a good idea in that case. However since this is a _first-steps_ tutorial, I'm not sure it's that relevant. I suppose there will be advanced users trying to do this tutorial though, so perhaps we can mention other ways of organising these folders, but then again it's a personal choice as to how one organises their files.

Reply:
I've added a note on organising folders in larger or more complex projects and opted to just use the `src` folder for this tutorial.